services/udp/service.go

```diff
@@ -153,15 +156,17 @@ func (s *Service) serve() {
 			return
 		default:
 			// Keep processing.
-			buf := make([]byte, s.config.UDPPayloadSize)
 			n, _, err := s.conn.ReadFromUDP(buf)
 			if err != nil {
 				s.statMap.Add(statReadFail, 1)
 				s.Logger.Printf("Failed to read UDP message: %s", err)
 				continue
 			}
 			s.statMap.Add(statBytesReceived, int64(n))
-			s.parserChan <- buf[:n]
+
+			bufCopy := make([]byte, n)
+			copy(bufCopy, buf[:n])
+			s.parserChan <- bufCopy
```

Comment:
This is correct the way it's written. `ReadFromUDP` can't modify our slice, so `buf` will always refer to the full 64K slice, and will get data copied into it starting at index zero, and return `n`, the number of bytes that were copied for a given read.